Feature/109 create a dynamodb resource in the cdk

hitohata opened the pull request:
## Overviews of implementation

Added DynamoDB to the CDK code.

## Review points

check if this satisfies the requirement.

hitohata committed:
feature: added helper functions and type.

hitohata committed:
feature: updated a comment in the helper function.

hitohata committed:
feature: added DynamoDB resource

hitohata committed:
chore: format

taniguchiiqqq reviewed (approved):
I checked if it can be deployed in my account. It succeeded. It doesn't have NonKeyAttributes and GSIs , but I understood this PR is about just creating dynamoDB resource from codes. so I assume it's what you intended.

hitohata committed:
feature: added GSIs

hitohata committed:
feature: updated a removal policy

hitohata committed:
chore: format

hitohata commented:
> I checked if it can be deployed in my account. It succeeded. It doesn't have NonKeyAttributes and GSIs , but I understood this PR is about just creating dynamoDB resource from codes. so I assume it's what you intended.

I'm sorry I completely overlooked it. I added those resources.
Also updated [the document](https://github.com/genesis-tech-tribe/nishiki-documents/pull/53).

taniguchiiqqq reviewed (approved):
I'm so sorry for really late review. I deployed the updated cdk and ensure the dynamoDB table has attributes and indexes which I saw in your json file before.
